portal/worker/src/walrus-sites-sw.ts

```diff
@@ -50,8 +54,40 @@ self.addEventListener("fetch", async (event) => {
     console.log("Parsed URL: ", parsedUrl);
 
     if (requestDomain == portalDomain && parsedUrl && parsedUrl.subdomain) {
-        console.log("fetching from the service worker");
-        event.respondWith(resolveAndFetchPage(parsedUrl));
+        event.respondWith((async () => {
+            // Clean the cache of expired entries
+            await cleanExpiredCache();
+            if (!('caches' in self)) {
+                // When not being in a secure context, the Cache API is not available.
+                console.warn('Cache API not available');
+                return await resolveAndFetchPage(parsedUrl);
+            }
+
+            const cache = await caches.open(cacheName);
+            const cachedResponse = await cache.match(urlString);
+            let isCacheSameAsNetwork: boolean;
+            try {
+                if (cachedResponse) {
+                    isCacheSameAsNetwork = await checkCachedVersionMatchesOnChain(cachedResponse)
+                }
+            } catch (e) {
+                console.error("Error checking cache version against chain:", e);
+            }
+            if (cachedResponse && isCacheSameAsNetwork) {
```

Comment:
Question: i think `isCacheSameAsNetwork` may be uninitialized at this point?

Reply:
It's true that there can be a case where the `isCacheSameAsNetwork` can be uninitialized, but this is not an issue in typescript, because uninitialized == `undefined`.

e.g. 
```typescript
let x: boolean;
x ? true : false; // Will return false, because boolean operations with `undefined` result to `false` in js/ts
```

Reply:
omg javascript 🙄